From 7b2a42ea8273bf588aa7c1f5b1d521d83d3b5038 Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Fri, 10 Jan 2025 14:50:00 +0100
Subject: [PATCH 763/841] refactor(stm32mp2): make
 stm32_pwr_cpu2_state_is_running MISRA compliant

Updates stm32_pwr_cpu2_state_is_running() to be static and MISRA
compliant using the __maybe_unused keyword on its parameter. This
ensures uniformity with stm32_pwr_cpu3_state_is_running().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git accb32eb39e96692cb05a08916179c8bf8b66b31 refactor(stm32mp2): make stm32_pwr_cpu2_state_is_running MISRA compliant]
Change-Id: If0ddca590224e19f6ed6db3a55a07496915f4cf9
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/428136
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 3138c77ed..f1853dd93 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -460,7 +460,7 @@ static bool stm32_freeze_other_core(unsigned int core_id)
 	return result;
 }
 
-bool stm32_pwr_cpu2_state_is_running(uintptr_t pwr_base)
+static bool stm32_pwr_cpu2_state_is_running(__maybe_unused uintptr_t pwr_base)
 {
 #if STM32MP_M33_TDCID
 	return true;
-- 
2.47.2

